src/test/kotlin/uk/gov/justice/digital/hmpps/approvedpremisesapi/integration/BedSearchRepositoryTest.kt

```diff
@@ -146,8 +146,8 @@ class BedSearchRepositoryTest : IntegrationTestBase() {
     bookingEntityFactory.produceAndPersist {
       withPremises(premisesOneMatchingEverything)
       withBed(bedTwoInRoomOneInPremisesOneMatchingEverything)
-      withArrivalDate(LocalDate.parse("2023-03-10"))
-      withDepartureDate(LocalDate.parse("2023-03-12"))
+      withArrivalDate(LocalDate.parse("2023-03-14"))
```

Comment:
If I recall our chat there is probably some useful context for the commit description here. Were these tests wrong prior to our work or as a result of a change in behaviour we've introduced?

Reply:
Updated commit to explain why these changes were made, but, to clarify:
I updated a slight error made regarding the `arrivalDate` and `departureDate of some instantiated beds and associated bookings within this commit. For instance, `bedThreeInRoomOneInPremisesOneMatchingEverything` was given the name "Matching bed with overlapping booking in middle", but its associated booking contained `arrivalDate` and `departureDate` values which indicated an overlapping booking spanning the end of the date range provided in `BedSearchRepository.findTemporaryAccommodationBeds`, not the middle.